builtin/credential/aws/path_role.go

```diff
@@ -742,29 +801,81 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
+	BoundIamPrincipalARNs       []string      `json:"bound_iam_principal_arn_list" mapstructure:"bound_iam_principal_arn_list"`
+	BoundIamPrincipalID         string        `json:"bound_iam_principal_id,omitempty" mapstructure:"bound_iam_principal_id"`
+	BoundIamPrincipalIDs        []string      `json:"bound_iam_principal_id_list" mapstructure:"bound_iam_principal_id_list"`
+	BoundIamRoleARN             string        `json:"bound_iam_role_arn,omitempty" mapstructure:"bound_iam_role_arn"`
+	BoundIamRoleARNs            []string      `json:"bound_iam_role_arn_list" mapstructure:"bound_iam_role_arn_list"`
+	BoundIamInstanceProfileARN  string        `json:"bound_iam_instance_profile_arn,omitempty" mapstructure:"bound_iam_instance_profile_arn"`
+	BoundIamInstanceProfileARNs []string      `json:"bound_iam_instance_profile_arn_list" mapstructure:"bound_iam_instance_profile_arn_list"`
+	BoundRegion                 string        `json:"bound_region,omitempty" mapstructure:"bound_region"`
+	BoundRegions                []string      `json:"bound_region_list" mapstructure:"bound_region_list"`
+	BoundSubnetID               string        `json:"bound_subnet_id,omitempty" mapstructure:"bound_subnet_id"`
+	BoundSubnetIDs              []string      `json:"bound_subnet_id_list" mapstructure:"bound_subnet_id_list"`
+	BoundVpcID                  string        `json:"bound_vpc_id,omitempty" mapstructure:"bound_vpc_id"`
+	BoundVpcIDs                 []string      `json:"bound_vpc_id_list" mapstructure:"bound_vpc_id_list"`
+	InferredEntityType          string        `json:"inferred_entity_type" mapstructure:"inferred_entity_type"`
+	InferredAWSRegion           string        `json:"inferred_aws_region" mapstructure:"inferred_aws_region"`
+	ResolveAWSUniqueIDs         bool          `json:"resolve_aws_unique_ids" mapstructure:"resolve_aws_unique_ids"`
+	RoleTag                     string        `json:"role_tag" mapstructure:"role_tag"`
+	AllowInstanceMigration      bool          `json:"allow_instance_migration" mapstructure:"allow_instance_migration"`
+	TTL                         time.Duration `json:"ttl" mapstructure:"ttl"`
+	MaxTTL                      time.Duration `json:"max_ttl" mapstructure:"max_ttl"`
+	Policies                    []string      `json:"policies" mapstructure:"policies"`
+	DisallowReauthentication    bool          `json:"disallow_reauthentication" mapstructure:"disallow_reauthentication"`
+	HMACKey                     string        `json:"hmac_key" mapstructure:"hmac_key"`
+	Period                      time.Duration `json:"period" mapstructure:"period"`
+	Version                     int           `json:"version" mapstructure:"version"`
+}
+
+func (r *awsRoleEntry) ToResponseData() map[string]interface{} {
+	responseData := map[string]interface{}{
+		"auth_type":                      r.AuthType,
+		"bound_ami_id":                   r.BoundAmiIDs,
```

Comment:
Although I see the reason as to why the `keys` in the response map to the older fields and the values to the newer, I am feeling a little off about it.

The other option that is bugging me is to return both the singular and slice keys with deprecation notice for the former.

@jefferai Thoughts?

Reply:
Never mind. Ignore this comment.